File: src/intellibitz/sted/actions/CopyAction.java
package intellibitz.sted.actions;

import intellibitz.sted.event.FontMapChangeEvent;
import intellibitz.sted.event.FontMapChangeListener;
import intellibitz.sted.ui.STEDWindow;
import intellibitz.sted.util.Resources;

import javax.swing.*;
import javax.swing.event.ListSelectionEvent;
import javax.swing.event.TableModelEvent;
import java.awt.event.ActionEvent;

public class CopyAction
        extends TableModelListenerAction
        implements FontMapChangeListener {
    public CopyAction() {
        super();
    }


    /**
     * This fine grain notification tells listeners the exact range of cells,
     * rows, or columns that changed.
     */
    public void tableChanged(TableModelEvent e) {
//        this.setEnabled(!this.getSelectedRows().isEmpty());
    }

    public void valueChanged(ListSelectionEvent e) {
        final ListSelectionModel listSelectionModel =
                (ListSelectionModel) e.getSource();
        setEnabled(listSelectionModel.getMinSelectionIndex() >= 0);
    }

    public void actionPerformed(ActionEvent e) {
        final STEDWindow stedWindow = getSTEDWindow();
        stedWindow.getDesktop()
                .addToClipboard(Resources.ENTRIES, copySelectedRows());
    }

    public void stateChanged(FontMapChangeEvent e) {
        setEnabled(!getSelectedRows().isEmpty());
    }

}

File: src/main/intellibitz/sted/actions/CopyAction.kt
package sted.actions

import sted.event.FontMapChangeEvent
import sted.event.FontMapChangeListener
import sted.io.Resources
import java.awt.event.ActionEvent
import javax.swing.ListSelectionModel
import javax.swing.event.ListSelectionEvent
import javax.swing.event.TableModelEvent

class CopyAction : TableModelListenerAction(), FontMapChangeListener {
    /**
     * This fine grain notification tells listeners the exact range of cells,
     * rows, or columns that changed.
     */
    override fun tableChanged(e: TableModelEvent) {
//        this.setEnabled(!this.getSelectedRows().isEmpty());
    }

    override fun valueChanged(e: ListSelectionEvent) {
        val listSelectionModel = e.source as ListSelectionModel
        isEnabled = listSelectionModel.minSelectionIndex >= 0
    }

    override fun actionPerformed(e: ActionEvent) {
        stedWindow.desktop.addToClipboard(Resources.ENTRIES, copySelectedRows())
    }

    override fun stateChanged(e: FontMapChangeEvent) {
        isEnabled = !selectedRows.isEmpty()
    }
}